src/main/java/ca/uhn/fhir/jpa/starter/CollectDataResourceProvider.java

```diff
@@ -0,0 +1,155 @@
+package ca.uhn.fhir.jpa.starter;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+import com.google.common.base.Charsets;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpHeaders;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.IdType;
+import org.hl7.fhir.r4.model.Measure;
+import org.hl7.fhir.r4.model.MeasureReport;
+import org.hl7.fhir.r4.model.Observation;
+import org.hl7.fhir.r4.model.Parameters;
+import org.hl7.fhir.r4.model.Reference;
+import org.hl7.fhir.r4.model.Resource;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.jpa.rp.r4.MeasureResourceProvider;
+import ca.uhn.fhir.rest.annotation.IdParam;
+import ca.uhn.fhir.rest.annotation.Operation;
+import ca.uhn.fhir.rest.annotation.OperationParam;
+import ca.uhn.fhir.rest.api.Constants;
+import ca.uhn.fhir.rest.api.EncodingEnum;
+
+public class CollectDataResourceProvider {
+    protected static CloseableHttpClient ourHttpClient;
+
+    @Autowired
+    MeasureResourceProvider measureResourceProvider;
+
+    FhirContext ctx = FhirContext.forR4();
+
+    @Operation(name = PlirConstants.OPERATION_COLLECT_DATA, idempotent = true, type = Measure.class)
+    public Parameters collectDataOperation(HttpServletRequest theServletRequest, @IdParam IdType theId,
+            @OperationParam(name = PlirConstants.PARAM_PERIOD_START, min = 1, max = 1) String periodStart,
+            @OperationParam(name = PlirConstants.PARAM_PERIOD_END, min = 1, max = 1) String periodEnd)
+            throws ClientProtocolException, IOException {
+   
+        Measure measure = this.measureResourceProvider.getDao().read(theId);
+ 
+        if (measure == null) {
+            throw new RuntimeException("Could not find Measure/" + theId.getIdPart());
+        }
+      
+        String fullUrl = constructUrl(theServletRequest, measure, periodStart, periodEnd);
+        Parameters parameters = new Parameters();
+        MeasureReport report = createMeasureReport(fullUrl);
+        Bundle observationBundle = createObsBundle(fullUrl);
+  
+        parameters.addParameter(new Parameters.ParametersParameterComponent()
+                .setName(PlirConstants.PARAMETER_NAME_MEASURE_REPORT).setResource(report));
+
+        parameters.addParameter(new Parameters.ParametersParameterComponent()
+                .setName(PlirConstants.PARAMETER_NAME_RESOURCE).setResource(observationBundle));
```

Comment:
We should be returning the individuals Obs (and Subjects) as separate resources rather than a bundle.

Something like:

```java
bundle.getEntry().stream()
				.filter(Bundle.BundleEntryComponent::hasResource)
				.map(Bundle.BundleEntryComponent::getResource)
				.forEach(r -> parameters.addParameter().setName(PlirConstants.PARAMETER_NAME_RESOURCE).setResource(r));
```